scripts/merge_config.sh

```diff
@@ -2,20 +2,69 @@
 
 set -e
 
-# optional env variable override
-if [ -n "$SYSREPOCFG_EXECUTABLE" ]; then
-    SYSREPOCFG="$SYSREPOCFG_EXECUTABLE"
-# avoid problems with sudo PATH
-elif [ `id -u` -eq 0 ] && [ -n "$USER" ] && [ `command -v su` ]; then
-    SYSREPOCFG=`su -c 'command -v sysrepocfg' -l $USER`
-else
-    SYSREPOCFG=`command -v sysrepocfg`
+if [ -z "$NP2_SCRIPTS_DIR" ]; then
+    echo "$0: Required environment variable NP2_SCRIPTS_DIR not set." >&2
+    exit 1
 fi
-KS_KEY_NAME=genkey
+
+# import functions
+source "${NP2_SCRIPTS_DIR}/common.sh"
+
+# get path to sysrepocfg executable, this will be stored in $SYSREPOCFG
+SYSREPOCFG_GET_PATH
 
 # check that there is no listen/Call Home configuration yet
-SERVER_CONFIG=`$SYSREPOCFG -X -x "/ietf-netconf-server:netconf-server/listen/endpoint[1]/name | /ietf-netconf-server:netconf-server/call-home/netconf-client[1]/name"`
-if [ -z "$SERVER_CONFIG" ]; then
+SERVER_CONFIG=$($SYSREPOCFG -X -x "/ietf-netconf-server:netconf-server/listen/endpoint | /ietf-netconf-server:netconf-server/call-home/netconf-client")
+if [ -n "$SERVER_CONFIG" ]; then
+    # the server is configured, just exit
+    exit 0
+fi
+
+# get the user who invoked the script
+CURRENT_USER="$SUDO_USER"
+# get his home dir
+CURRENT_USER_HOME=$(eval echo "~$CURRENT_USER")
+# try to get his authorized_keys file
+AUTHORIZED_KEYS_FILE="$CURRENT_USER_HOME/.ssh/authorized_keys"
+# check if the authorized keys file exists
+if [ -f "$AUTHORIZED_KEYS_FILE" ]; then
+    # it exists, create public keys that are authorized in the server's configuration
+    AUTH_CONFIG="
+                        <public-keys>
+                            <inline-definition>"
+
+    IDX=0
+# read lines from authorized_keys
+    while IFS= read -r LINE; do
+        # check if the line is empty or starts with a comment (#)
+        if [[ -n "$LINE" && ! "$LINE" =~ ^\s*# ]]; then
+            # extract the base64 public key
+            PUB_BASE64=$(echo "$LINE" | awk '{print $2}')
+
+            NEW_PUBKEY_ENTRY="  <public-key>
+                                    <name>authorized_key_${IDX}</name>
+                                    <public-key-format xmlns:ct=\"urn:ietf:params:xml:ns:yang:ietf-crypto-types\">ct:ssh-public-key-format</public-key-format>
+                                    <public-key>${PUB_BASE64}</public-key>
+                                </public-key>"
+            # append
+            AUTH_CONFIG="${AUTH_CONFIG}${NEW_PUBKEY_ENTRY}"
+            IDX=$((IDX + 1))
+        fi
+    done < "$AUTHORIZED_KEYS_FILE"
+
+    # append the ending tags
+    AUTH_CONFIG="${AUTH_CONFIG}
+                            </inline-definition>
+                        </public-keys>"
+
+    echo "Creating user \"${CURRENT_USER}\" that can authenticate with a key pair from his authorized_keys..."
```

Comment:
The message (and the one below) is a bit inaccurate, the user is not being created, only added into the server configuration.